funsor/adjoint.py

```diff
@@ -56,7 +56,8 @@ def adjoint(self, red_op, bin_op, root, targets):
 
         bin_unit = to_funsor(ops.UNITS[bin_op])
         adjoint_values = defaultdict(lambda: bin_unit)
-        adjoint_values[root] = root
+        # TODO should this be root or not?
+        # adjoint_values[root] = root
```

Comment:
With the `adj[root]=root` in `forward_backward` I had to subtract(divide) twice `Z` to get the marginal which is an unnecessary extra computation if the goal is to get marginals:

```py
        # adjoint returns Z * dZ / df = Z * alpha[t-1] * beta[t]
        # marginal = adj * f / Z / Z
        actual = ops.PRODUCT_INVERSES[prod_op](ops.PRODUCT_INVERSES[prod_op](prod_op(adj, f), Z), Z)
```